vertical-pod-autoscaler/pkg/admission-controller/logic/resource_handler.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logic
+
+import (
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/metrics/admission"
+)
+
+// ResourceHandler represents a handler for a resource in Admission Server
```

Comment:
Please describe this interface a bit more. After reading the interface I don't know how I would use it. `GroupResourceType` and `AdmissionResource` (why one is `ResourceType` but the other is just `Resource`?)  tell me what resource types it accepts? `DismissIncorrectObjects` tells me whether it will discard invalid objects or not? `GetPatches` actually does the work described by other methods?

Reply:
You got everything right :). Clarified in the comments. Does it look better now?